### PR TITLE
release one tispark jar instead of two 

### PR DESCRIPTION
**Pull request by marsishandsome**
close https://github.com/pingcap/tispark/issues/932

### What problem does this PR solve? 
currently we must release two jars, one for spark-2.3 another for spark-2.4

better to release only one jar, both work for spark-2.3 and spark-2.4

### What is changed and how it works?
1. create two new projects `spark-wrapper-2.3` and `spark-wrapper-2.4` to place code with compatibility issues.
2. create a new project `tispark-assembly` to package `spark-wrapper-2.3` and `spark-wrapper-2.4` into release jar under `resources` instead of default classpath
3. according to runtime spark version, tispark will use classloader to load the correct version of the `spark-wrapper`

### Related changes
 - Need to update the `tidb-ansible` repository (jar name change from tispark-core to tispark-assembly)  https://github.com/pingcap/tidb-ansible/pull/833
- Need to update the `tidb-docker-compose` repository https://github.com/pingcap/tispark/issues/935
- Need to update document  (dependency tispark-core -> tispark-assembly) https://github.com/pingcap/tispark/issues/936
 
### TO Check:
- [x] run tispark project example in IDEA with spark-2.3.3 & spark-2.4.3
- [x] run tispark project ut in IDEA with spark-2.3.3 & spark-2.4.3
- [x] run tispark user example in IDEA with spark-2.3.3 & spark-2.4.3
- [x] run tispark user example by spark-submit (package without tispark jar) with spark-2.3.3 & spark-2.4.3
- [x] run tispark user example by spark-submit (package with tispark jar) with spark-2.3.3 & spark-2.4.3
- [x] run ut in github ci with spark-2.4.3 https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tispark_ghpr_integration_test/detail/tispark_ghpr_integration_test/704/pipeline/
- [x] run ut in github ci with spark-2.3.3  https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tispark_ghpr_integration_test/detail/tispark_ghpr_integration_test/705/pipeline

test example: https://github.com/pingcap/tispark-test/blob/master/tispark-examples/src/main/scala/com/pingcap/tispark/examples/OneJarExample.scala

### CODE REVIEW

**Comment by birdstorm** [comment]
How large is the jar size now?

**Comment by marsishandsome** [comment]
> How large is the jar size now?

28M

File: assembly/pom.xml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            <groupId>com.pingcap.tispark</groupId>
+            <artifactId>tispark-core</artifactId>
+            <version>${project.parent.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.pingcap.tikv</groupId>
+            <artifactId>tikv-client</artifactId>
+            <version>${project.parent.version}</version>
+        </dependency>
+        <dependency>
```
[the remark this answers is not in the document]
Comment (marsishandsome) [reply assembly/pom.xml]: yes, user can use tispark without adding mysql-jdbc connector.

[the remark this answers is not in the document]
Comment (zhexuany) [reply assembly/pom.xml]: If that is the case, we need inform some of our customers.

[the remark this answers is not in the document]
Comment (marsishandsome) [reply assembly/pom.xml]: nice point

File: tikv-client/pom.xml [hunk 2]
```diff
@@ -308,9 +308,6 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
                 <version>2.9.1</version>
-                <configuration>
-                    <skip>${javadoc.skip}</skip>
```
Comment (birdstorm) [inline tikv-client/pom.xml]: does it mean that we will generate javadoc every time after compile?

Comment (marsishandsome) [reply tikv-client/pom.xml]: fixed

**Pull request merged**